cumulus/pallets/collator-selection/src/lib.rs

```diff
@@ -183,8 +189,8 @@ pub mod pallet {
 	/// The (community, limited) collation candidates. `Candidates` and `Invulnerables` should be
 	/// mutually exclusive.
 	#[pallet::storage]
-	#[pallet::getter(fn candidates)]
-	pub type Candidates<T: Config> = StorageValue<
+	#[pallet::getter(fn candidate_list)]
+	pub type CandidateList<T: Config> = StorageValue<
```

Comment:
here we should document in which order the elements stored in the list.

and I think we do not really need `take_candidate_slot`. we can always maintain a sorted list, and when new candidate registered with `register_as_candidate`, it is inserted into the right position instead to `0`. So we will need only `update_bond`.

Reply:
also your `update_bond` does not with for candidate registered at index `0`.

Reply:
we need a doc here saying in which order the elements stored in the list.

Reply:
> I think we do not really need `take_candidate_slot`. we can always maintain a sorted list, and when new candidate registered with `register_as_candidate`

There were multiple ways to solve this issue and we went with the `take_candidate_slot` route. What you describe would have also solved the problem, but I believe it's better UX with `take_candidate_slot` because the extrinsic itself is simpler to understand and to use from a user POV, it hides away less complexity and it preserves the already existing `register_candidate`. Also, it makes sense conceptually that you would register with the minimum bid in the auction anyway until all the spots are taken, which is what `register_candidate` does now. Technically you can put the `take_candidate_slot` logic into `register_as_candidate`, but I prefer them to be separate.

> also your `update_bond` does not with for candidate registered at index 0.

What do you mean?

> we need a doc here saying in which order the elements stored in the list.

Sure.

Reply:
discussed offline. I am convinced with the approach.
the list might be not ordered only if deposit will be set to higher value than before.